### PR TITLE
Test thunked tangents

### PR DESCRIPTION
**Pull request by mzgubic**
With the 1.0 release of ChainRules ecosystem (coming up soon), rules will need to support receiving thunked tangents. This PR expands the tests for all the rules to include receiving a thunked tangent.

### CODE REVIEW

**Comment by stevengj** [comment]
LGTM.

File: test/runtests.jl
Comment (stevengj) [inline test/runtests.jl:40]: This seems like an undocumented API…

If this feature is going to be required in ChainRules.jl, shouldn't ChainRulesTestUtils.jl implement a documented API to test it?

Comment (oxinabox) [reply test/runtests.jl:40]: A comment  should be added here to explain what is going on
(and perhaps more clear docs, it is mentioned [here](https://juliadiff.org/ChainRulesTestUtils.jl/dev/api.html#ChainRulesTestUtils.test_rrule-Tuple) but not about mutating it).

This is the soft-way of rolling out the changes.
ChainRulesTestUtils v1.0 will define 
`ChainRulesTestUtils.TRANSFORMS_TO_ALT_TANGENTS = [x -> @thunk(x), x->ZeroTangent(), ...]`.
But right now it is set to `[]`.
And then as we go and fix packages we can add this line to make sure it gets those new tests.
So that we can roll that out one package at a time, without breaking every other package.

In a normal package this would be unsafe, since each package mutating it could be seen by each other.
But ChainRulesTestUtils is only ever a direct, test-time, dependency.
So I believe it is safe.

The 1.0 API will mean this doesn't need to be touched and this line will have no effect.

Comment (mzgubic) [reply test/runtests.jl:40]: I came here to say what Lyndon said. Will add a comment explaining this temporary state

**Comment by mzgubic** [comment]
Added the comment. I am not in JuliaMath so don't have merge rights, please feel free to merge whenever.

File: test/runtests.jl
Comment (stevengj) [reply test/runtests.jl:45]: I agree that this test should be opt-in, but it seems like it would be cleaner if you could just opt-in by something like:
```jl
ChainRulesTestUtils.enable_thunk_testing()
```
and this would become a no-op in 1.0.

Comment (oxinabox) [reply test/runtests.jl:45]: That's true.
I will make that change and then we can update this

**Review by stevengj — APPROVED** [review]

**Comment by oxinabox** [comment]
Alt could just not merge it.
It is not really a functional change.
It is just proving that the tests for 1.0.0 will pass.
and we will release 1.0.0 soon.
So that should be fine to just not merge this and merge that one instead.

**Pull request closed without merging**
